### PR TITLE
Alignment enum

### PR DESCRIPTION
**Pull request by giladgray**
- add `Alignment` enum and `Classes.alignmentClass()`

### CODE REVIEW

File: packages/core/src/components/button/abstractButton.tsx [hunk 1]
```diff
@@ -26,6 +26,8 @@ export interface IButtonProps extends IActionProps {
      * Text alignment within button. By default, icons and text will be centered within the button.
      * Passing this prop will cause the text container to fill the button and align the text within that
      * to the appropriate side. `icon` and `rightIcon` will be pushed to either side.
+     *
+     * The `Alignment` enum provides constants for allowed values.
```
Comment (adidahiya) [inline packages/core/src/components/button/abstractButton.tsx]: then why not use the enum here?

File: packages/core/src/components/button/buttonGroup.tsx [hunk 2]
```diff
@@ -15,6 +15,8 @@ export interface IButtonGroupProps extends IProps, React.HTMLProps<HTMLDivElemen
      * `align="left"` will left-align button text and push `rightIcon` to right side.
      * `align="right"` right-aligns text and pushes `icon` to left side.
      * This prop only has an effect if buttons are wider than their default widths.
+     *
+     * The `Alignment` enum provides constants for allowed values.
```
Comment (adidahiya) [inline packages/core/src/components/button/buttonGroup.tsx]: same

File: packages/core/src/components/navbar/navbarGroup.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { IProps } from "../../common/props";
 export interface INavbarGroupProps extends React.HTMLProps<HTMLDivElement>, IProps {
     /**
      * The side of the navbar on which the group should appear.
+     * The `Alignment` enum provides constants for these values.
```
Comment (adidahiya) [inline packages/core/src/components/navbar/navbarGroup.tsx]: same

File: packages/core/src/common/alignment.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/*
+* Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+*
+* Licensed under the terms of the LICENSE file distributed with this project.
+*/
+
+export type AlignmentType = "center" | "left" | "right";
```
Comment (adidahiya) [inline packages/core/src/common/alignment.ts]: why do you need this in addition to the enum?

File: packages/core/src/components/button/abstractButton.tsx (hunk 1 above)
Comment (giladgray) [reply packages/core/src/components/button/abstractButton.tsx]: like #2090, i want to show the actual options in the docs, not the type alias name.

File: packages/core/src/components/button/buttonGroup.tsx (hunk 2 above)
Comment (giladgray) [reply packages/core/src/components/button/buttonGroup.tsx]: string literals are not assignable to enum types (we learned this with string enums #1921)

**Comment by giladgray** [comment]
@adidahiya we learned in #1921 that `"primary"` is not assignable to enum type `Intent` so I decided to try a different approach to typing a feature like that. hence the separation between enum and union type. i created union type for use internally, but decided not to use it for public component props so the docs will show the three accepted values instead of the inscrutable `AlignmentType`.

**Comment by adidahiya** [comment]
If you really want both a type and an enum, you have to do something like this:

```ts
export type Alignment = "left" | "center" | "right";
export const Alignment = {
    LEFT: "left" as Alignment,
    CENTER: "center" as Alignment,
    RIGHT: "right" as ALIGNMENT,
};
```

... so we should either do that (overload one symbol) or do nothing at all

**Comment by giladgray** [comment]
I think you mean `as "left"`, etc.

**Comment by giladgray** [comment]
@adidahiya hmm yeah thoughts on migrating `Position` and `Intent` to work like that?

**Pull request merged**